### ISSUE TITLE
Automatic redirect not working without domain assigned

### ISSUE THREAD

**Issue by pluraliscode** [kipusoep/UrlTracker#65]
I have installed version 3.3 on a live site, and everything seems to work fine - except the redirect is not taking place. Users are redirected to the 404 page instead.

This seems to be because there is no domain name assigned to the root node of the pages that are not redirected. Since the root-nodes are being used differently than in a normal tree (most are used as 'folders' to keep the admin simple) I cannot set a domain name on each - since they are all used for the same site.

Is there any way to use UrlTracker with multiple root nodes - where only one has an assigned hostname? Or in other words for nodes under a root node without a hostname?

**Comment by kipusoep** [comment]
Have you tried v3.5?

**Comment by pluraliscode** [comment]
Yes, just did - it still doesn't redirect, but url changes are registered correctly it seems.

**Comment by kipusoep** [comment]
Could you tell me what the output of the logs is when you enable it and request an 'Old URL'?

Instructions to enable logging:
* Add an appSetting to your web.config: ```<add key="urlTracker:enableLogging" value="true"/>```
* In 'config\log4net.config', change the root logger priority to 'Debug' instead of 'Info': ```<priority value="Debug"/>```

The lines I'm interested in contain "InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper".

**Comment by pluraliscode** [comment]
Yes, the log contains the following:

2015-03-27 05:14:06,988 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| AcquireRequestState start
2015-03-27 05:14:06,988 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 35]
UrlTracker HttpModule | AcquireRequestState start
2015-03-27 05:14:06,988 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| Incoming URL is: website/filmblog-wt21.aspx
2015-03-27 05:14:06,988 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 35]
UrlTracker HttpModule | Incoming URL is: website/filmblog-wt21.aspx
2015-03-27 05:14:06,988 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| Response statusCode is 404, continue URL matching
2015-03-27 05:14:06,988 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 35]
UrlTracker HttpModule | Response statusCode is 404, continue URL matching
2015-03-27 05:14:06,988 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| Current request's rootNodeId: -1
2015-03-27 05:14:06,988 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 35]
UrlTracker HttpModule | Current request's rootNodeId: -1
2015-03-27 05:14:07,003 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| No match found in AcquireRequestState
2015-03-27 05:14:07,003 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 35]
UrlTracker HttpModule | No match found in AcquireRequestState
2015-03-27 05:14:07,003 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| AcquireRequestState end
2015-03-27 05:14:07,003 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 35]
UrlTracker HttpModule | AcquireRequestState end
2015-03-27 05:14:07,019 [36] DEBUG umbraco.UmbracoDefault - [Thread 35]
PreInit
2015-03-27 05:14:07,035 [36] DEBUG umbraco.UmbracoDefault - [Thread 35]
Complete (took 3ms)
2015-03-27 05:14:07,050 [36] DEBUG umbraco.UmbracoDefault - [Thread 35] Init
2015-03-27 05:14:07,050 [36] DEBUG umbraco.UmbracoDefault - [Thread 35]
Complete (took 0ms)
2015-03-27 05:14:07,050 [36] DEBUG umbraco.UmbracoDefault - [Thread 35] Load
2015-03-27 05:14:07,066 [36] DEBUG umbraco.UmbracoDefault - [Thread 35]
Complete (took 9ms)
2015-03-27 05:14:07,066 [36] DEBUG umbraco.UmbracoDefault - [Thread 35]
Render
2015-03-27 05:14:07,066 [36] DEBUG umbraco.UmbracoDefault - [Thread 35]
Complete (took 3ms)
2015-03-27 05:14:07,066 [36] DEBUG Umbraco.Web.UmbracoModule - [Thread 35]
Total milliseconds for umbraco request to process: 671.875
2015-03-27 05:14:07,066 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| EndRequest start
2015-03-27 05:14:07,066 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 35]
UrlTracker HttpModule | EndRequest start
2015-03-27 05:14:07,066 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| Incoming URL is: website/filmblog-wt21.aspx
2015-03-27 05:14:07,066 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 35]
UrlTracker HttpModule | Incoming URL is: website/filmblog-wt21.aspx
2015-03-27 05:14:07,066 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| Response statusCode is 404, continue URL matching
2015-03-27 05:14:07,066 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 35]
UrlTracker HttpModule | Response statusCode is 404, continue URL matching
2015-03-27 05:14:07,066 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| Current request's rootNodeId: -1
2015-03-27 05:14:07,066 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 35]
UrlTracker HttpModule | Current request's rootNodeId: -1
2015-03-27 05:14:07,082 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| No match found, logging as 404 not found
2015-03-27 05:14:07,082 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 35]
UrlTracker HttpModule | No match found, logging as 404 not found
2015-03-27 05:14:07,082 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| EndRequest end
2015-03-27 05:14:07,082 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 35]
UrlTracker HttpModule | EndRequest en
2015-03-27 05:14:11,191 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| AcquireRequestState start
2015-03-27 05:14:11,191 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 58]
UrlTracker HttpModule | AcquireRequestState start
2015-03-27 05:14:11,191 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| Incoming URL is:
umbraco/webservices/legacyAjaxCalls.asmx/GetSecondsBeforeUserLogout
2015-03-27 05:14:11,191 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 58]
UrlTracker HttpModule | Incoming URL is:
umbraco/webservices/legacyAjaxCalls.asmx/GetSecondsBeforeUserLogout
2015-03-27 05:14:11,191 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| Checking for forced redirects (AcquireRequestState), continue URL matching
2015-03-27 05:14:11,191 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 58]
UrlTracker HttpModule | Checking for forced redirects
(AcquireRequestState), continue URL matching
2015-03-27 05:14:11,191 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| URL is an umbraco reserved path or url, ignore request
2015-03-27 05:14:11,191 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 58]
UrlTracker HttpModule | URL is an umbraco reserved path or url, ignore
request
2015-03-27 05:14:11,191 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| EndRequest start
2015-03-27 05:14:11,191 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 58]
UrlTracker HttpModule | EndRequest start
2015-03-27 05:14:11,191 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| Incoming URL is:
umbraco/webservices/legacyAjaxCalls.asmx/GetSecondsBeforeUserLogout
2015-03-27 05:14:11,191 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 58]
UrlTracker HttpModule | Incoming URL is:
umbraco/webservices/legacyAjaxCalls.asmx/GetSecondsBeforeUserLogout
2015-03-27 05:14:11,191 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| Response statuscode is not 404, UrlTracker won't do anything
2015-03-27 05:14:11,191 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 58]
UrlTracker HttpModule | Response statuscode is not 404, UrlTracker won't do
anything
2015-03-27 05:14:11,191 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| EndRequest end
2015-03-27 05:14:11,191 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 58]
UrlTracker HttpModule | EndRequest end
2015-03-27 05:14:21,238 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| AcquireRequestState start
2015-03-27 05:14:21,238 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 35]
UrlTracker HttpModule | AcquireRequestState start
2015-03-27 05:14:21,238 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| Incoming URL is:
umbraco/webservices/legacyAjaxCalls.asmx/GetSecondsBeforeUserLogout
2015-03-27 05:14:21,238 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 35]
UrlTracker HttpModule | Incoming URL is:
umbraco/webservices/legacyAjaxCalls.asmx/GetSecondsBeforeUserLogout
2015-03-27 05:14:21,238 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| Checking for forced redirects (AcquireRequestState), continue URL matching
2015-03-27 05:14:21,238 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 35]
UrlTracker HttpModule | Checking for forced redirects
(AcquireRequestState), continue URL matching
2015-03-27 05:14:21,238 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| URL is an umbraco reserved path or url, ignore request
2015-03-27 05:14:21,238 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 35]
UrlTracker HttpModule | URL is an umbraco reserved path or url, ignore
request
2015-03-27 05:14:21,253 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| EndRequest start
2015-03-27 05:14:21,253 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 35]
UrlTracker HttpModule | EndRequest start
2015-03-27 05:14:21,253 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| Incoming URL is:
umbraco/webservices/legacyAjaxCalls.asmx/GetSecondsBeforeUserLogout
2015-03-27 05:14:21,253 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 35]
UrlTracker HttpModule | Incoming URL is:
umbraco/webservices/legacyAjaxCalls.asmx/GetSecondsBeforeUserLogout
2015-03-27 05:14:21,253 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| Response statuscode is not 404, UrlTracker won't do anything
2015-03-27 05:14:21,253 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 35]
UrlTracker HttpModule | Response statuscode is not 404, UrlTracker won't do
anything
2015-03-27 05:14:21,253 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| EndRequest end
2015-03-27 05:14:21,253 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 35]
UrlTracker HttpModule | EndRequest end
2015-03-27 05:14:31,238 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| AcquireRequestState start
2015-03-27 05:14:31,238 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 58]
UrlTracker HttpModule | AcquireRequestState start
2015-03-27 05:14:31,238 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| Incoming URL is:
umbraco/webservices/legacyAjaxCalls.asmx/GetSecondsBeforeUserLogout
2015-03-27 05:14:31,238 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 58]
UrlTracker HttpModule | Incoming URL is:
umbraco/webservices/legacyAjaxCalls.asmx/GetSecondsBeforeUserLogout
2015-03-27 05:14:31,238 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| Checking for forced redirects (AcquireRequestState), continue URL matching
2015-03-27 05:14:31,238 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 58]
UrlTracker HttpModule | Checking for forced redirects
(AcquireRequestState), continue URL matching
2015-03-27 05:14:31,238 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| URL is an umbraco reserved path or url, ignore request
2015-03-27 05:14:31,238 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 58]
UrlTracker HttpModule | URL is an umbraco reserved path or url, ignore
request
2015-03-27 05:14:31,238 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| EndRequest start
2015-03-27 05:14:31,238 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 58]
UrlTracker HttpModule | EndRequest start
2015-03-27 05:14:31,238 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| Incoming URL is:
umbraco/webservices/legacyAjaxCalls.asmx/GetSecondsBeforeUserLogout
2015-03-27 05:14:31,238 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 58]
UrlTracker HttpModule | Incoming URL is:
umbraco/webservices/legacyAjaxCalls.asmx/GetSecondsBeforeUserLogout
2015-03-27 05:14:31,238 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| Response statuscode is not 404, UrlTracker won't do anything
2015-03-27 05:14:31,238 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 58]
UrlTracker HttpModule | Response statuscode is not 404, UrlTracker won't do
anything
2015-03-27 05:14:31,238 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| EndRequest end
2015-03-27 05:14:31,238 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 58]
UrlTracker HttpModule | EndRequest end
2015-03-27 05:14:33,957 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| AcquireRequestState start
2015-03-27 05:14:33,957 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 14]
UrlTracker HttpModule | AcquireRequestState start
2015-03-27 05:14:33,957 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| Incoming URL is: umbraco/RestServices/ScheduledPublish/Index
2015-03-27 05:14:33,957 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 14]
UrlTracker HttpModule | Incoming URL is:
umbraco/RestServices/ScheduledPublish/Index
2015-03-27 05:14:33,957 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| Checking for forced redirects (AcquireRequestState), continue URL matching
2015-03-27 05:14:33,957 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 14]
UrlTracker HttpModule | Checking for forced redirects
(AcquireRequestState), continue URL matching
2015-03-27 05:14:33,957 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| URL is an umbraco reserved path or url, ignore request
2015-03-27 05:14:33,957 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 14]
UrlTracker HttpModule | URL is an umbraco reserved path or url, ignore
request
2015-03-27 05:14:34,035 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| EndRequest start
2015-03-27 05:14:34,035 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 14]
UrlTracker HttpModule | EndRequest start
2015-03-27 05:14:34,035 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| Incoming URL is: umbraco/RestServices/ScheduledPublish/Index
2015-03-27 05:14:34,035 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 14]
UrlTracker HttpModule | Incoming URL is:
umbraco/RestServices/ScheduledPublish/Index
2015-03-27 05:14:34,035 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| Response statuscode is not 404, UrlTracker won't do anything
2015-03-27 05:14:34,035 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 14]
UrlTracker HttpModule | Response statuscode is not 404, UrlTracker won't do
anything
2015-03-27 05:14:34,035 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - UrlTracker HttpModule
| EndRequest end
2015-03-27 05:14:34,035 [36] DEBUG
InfoCaster.Umbraco.UrlTracker.Helpers.LoggingHelper - [Thread 14]
UrlTracker HttpModule | EndRequest end


hope it helps

**Comment by kipusoep** [comment]
Thanks! What does the rule look like, can you post a screenshot of your content tree and the redirect in the UrlTracker view?

**Comment by pluraliscode** [comment]
Yes - here is the rule

​and the content tree

Hope it helps
​

**Comment by kipusoep** [comment]
Where?

**Comment by pluraliscode** [comment]
Sorry here - 
![screen shot 2015-04-08 at 15 59 10](https://cloud.githubusercontent.com/assets/4352422/7047994/600c67d4-de10-11e4-813d-cc10e655d3fd.png)

and 
![screen shot 2015-04-08 at 16 02 34](https://cloud.githubusercontent.com/assets/4352422/7048018/73c0e232-de10-11e4-8bc5-179283ef0a31.png)

**Comment by kipusoep** [comment]
Could you try again with v3.8, which I just released?

**Comment by pluraliscode** [comment]
Yes, I have uninstalled the old package, installed version 3.8, tested one
of the old redirects from earlier tests, tested a new automatically
generated redirect. Without luck unfortunately.

**Comment by kipusoep** [comment]
Would it be possible to share the website with me, so I can attach a debugger and find out what's happening?

**Comment by pluraliscode** [comment]
Yes, I can either send you a copy of the site or give you access to
Umbraco. Which do you prefer and can i PM you?

**Comment by kipusoep** [comment]
I'll have to debug, so I need a copu of the site + the database.
You can contact me via stefan.kip [at] infocaster [dot] net, thanks in advance.
